src/Common.am

```diff
@@ -73,3 +74,21 @@ COMPAT_LIB = $(top_builddir)/compat/libcompatsquid.la $(LIBPROFILER)
 
 ## Some helpers are written in Perl and need the local shell defined properly
 subst_perlshell = sed -e 's,[@]PERL[@],$(PERL),g' <$(srcdir)/$@.pl.in >$@ || ($(RM) -f $@ ; exit 1)
+
+EXTENSIONS=.hdrtest
+.h.hdrtest:
+	@srcfilepath=`echo $< | $(SED) 's%^$(top_srcdir)/%%'`; \
+	srcfile=`basename $$srcfilepath`; \
+	tgtfile=`basename $@`; \
+	if test "" != "$(DONOTTEST)" && echo "$(DONOTTEST)" | $(TR) ' ' '\n' | $(GREP) -x -q "$$srcfile" ; then \
+		echo "skip - $$srcfilepath"; \
```

Comment:
After looking at your [sample](https://github.com/squid-cache/squid/pull/1463#issuecomment-1731419501) output, I suggest adding test name to all `echo` commands in this target:

```suggestion
		echo "hdrtest: skip - $$srcfilepath"; \
```

or, more human-friendly:

```suggestion
		echo "header-test: skip - $$srcfilepath"; \
```

... and the same for the other two `echo`s below, of course.